cache/remote.go

```diff
@@ -313,10 +313,22 @@ func (p lazyRefProvider) Info(ctx context.Context, dgst digest.Digest) (content.
 	if dgst != p.desc.Digest {
 		return content.Info{}, errdefs.ErrNotFound
 	}
-	if err := p.Unlazy(ctx); err != nil {
-		return content.Info{}, errdefs.ErrNotFound
+	info, err := p.ref.cm.ContentStore.Info(ctx, dgst)
```

Comment:
I changed the behavior of this function as `worker.FromRemote()` and cache marshaling uses this function to detect that content is available without opening reader or unlazy.

@ktock Looks like a stargz test broke so I think this was somehow used. Is it required for the unlazy to happen here (or maybe the test issue is something else)? The function was added in https://github.com/moby/buildkit/pull/4558/files#diff-c01e7be139ae4532b9aeef7b932fe60d3eb81680a7f4669b9dd0992ce71284f5R312 so I'm surprised it has this effect.